lib/compat/wordpress-6.5/class-block-search-controller.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Block search controller.
+ *
+ * @package Block_Search_Controller
+ */
+
+if ( ! class_exists( 'Block_Search_Controller' ) ) {
+	/**
+	 * Class Block_Search_Controller.
+	 */
+	class Block_Search_Controller extends WP_REST_Controller {
+
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+			$this->namespace = 'wp/v2';
+			$this->rest_base = 'block_search';
+		}
+
+		/**
+		 * Registers the routes for the objects of the controller.
+		 */
+		public function register_routes() {
+			register_rest_route(
+				$this->namespace,
+				'/' . $this->rest_base,
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => array(
+							$this,
+							'find_posts_by_block_attribute',
+						),
+						'args'                => array(
+							'post_type'       => array(
+								'required' => true,
+							),
+							'block_name'      => array(
+								'required' => true,
+							),
+							'attribute_name'  => array(
+								'required' => true,
+							),
+							'attribute_value' => array(
+								'required' => true,
+							),
+							'posts_per_page'  => array(
+								'required' => false,
+							),
```

Comment:
given how this is used it seems a bit misleading to call this `posts_per_page` when there's no mechanism to grab successive pages and when the behavior is closer to `max_posts` or `max_search_results`

Reply:
Good point.